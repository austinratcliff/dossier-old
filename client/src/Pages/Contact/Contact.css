.Contact {
  position: absolute;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background-color: white;
  color: #111;
  font-family: 'IBM Plex Mono', monospace;
  height: 100vh;
  width: 100vw;
  overflow: hidden;
}

.Contact .header {
  position: absolute;
  font-size: calc(10px + 5vmin);
  margin: 0;
  animation: slide-down 1.5s 1.5s 1 forwards;
}

.Contact .message {
  transform: translate(-5vw, 0);
  font-size: calc(10px + 2vmin);
  margin: 0;
  opacity: 0;
  visibility: hidden;
  animation: fade-in-right 1s 2s 1 forwards;
}

.Contact .contacts {
  transform: translate(5vw, 0);
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 30px 0 0 0;
  opacity: 0;
  visibility: hidden;
  animation: fade-in-left 1s 2s 1 forwards;
}

.Contact .icon {
  margin: 0 10px 0 20px;
}

.Contact .contact {
  color: #111;
  font-size: calc(10px + 1vmin);
  text-decoration: none;
}

.Contact .highlight {
  background: linear-gradient(180deg, rgba(255, 255, 255, 0) 50%, #92A8D1 50%);
  padding: 0 5px;
}

.Contact .highlight-hover:hover {
  background: linear-gradient(180deg, rgba(255, 255, 255, 0) 50%, #92A8D1 50%);
}

.Contact .home {
  position: absolute;
  top: 0;
  margin-top: 25px;
  cursor: pointer;
  animation: bounce-up 1s infinite alternate;
}

.Contact .home:hover {
  animation-play-state: paused;
}

@keyframes slide-down {
  from {transform: translateY(0);}
  to {transform: translateY(100vh);}
}

@keyframes fade-in-right {
  from {transform: translate(-5vw, 0); opacity: 0; visibility: hidden;}
  to {transform: translate(0); opacity: 1; visibility: visible;}
}

@keyframes fade-in-left {
  from {transform: translate(5vw, 0); opacity: 0; visibility: hidden;}
  to {transform: translate(0); opacity: 1; visibility: visible;}
}

@keyframes bounce-up {
  from {margin-top: 25px;}
  to {margin-top: 15px;}
}

/*
TODO: Add mobile responsive design
*/
