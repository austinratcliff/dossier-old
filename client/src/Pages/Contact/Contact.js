import React from 'react'
import './Contact.css'

const Contact = props => {
  return (
    <div className="Contact">
      <h1 className="header">
        <i
          className="fas fa-comment-dots"
          title="contact"
        /> Contact
      </h1>
      <i
        className="fas fa-angle-down fa-2x home"
        title="home"
        onClick={props.handleClick}
      />
    </div>
  )
}

export default Contact
