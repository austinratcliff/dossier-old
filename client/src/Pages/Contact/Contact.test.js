import React from 'react'
import ReactDOM from 'react-dom'
import ReactTestRenderer from 'react-test-renderer'
import Contact from './Contact'

describe('Contact component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Contact />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('matches snapshot', () => {
    const component = ReactTestRenderer.create(<Contact />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
