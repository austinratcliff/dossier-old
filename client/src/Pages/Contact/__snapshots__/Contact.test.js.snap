// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact component matches snapshot 1`] = `
<div
  className="Contact"
>
  <h1
    className="header"
  >
    <i
      className="fas fa-comment-dots"
      title="contact"
    />
     Contact
  </h1>
  <i
    className="fas fa-angle-down fa-2x home"
    title="home"
  />
</div>
`;
